src/components/ui/Animation.js

```diff
@@ -85,7 +85,7 @@ export default function animateChart(stx, animationParameters, easeMachine) {
         chart = chart || this.chart;
         if (chart.lockScroll) {
             const lastAppendQuote     = appendQuotes[appendQuotes.length - 1];
-            const lastDataSegmentItem = chart.dataSegment[chart.dataSegment.length - 1];
+            const lastDataSegmentItem = chart.dataSegment ? chart.dataSegment[chart.dataSegment.length - 1] : null;
```

Comment:
@morteza-binary I am not sure about this, why this happen sometimes

Reply:
@shayanbinary because it's the chart is reinitialized and there isn't any data on `dataSegment`